### PR TITLE
[improve][doc] Support configurable transactionBufferClientOperationTimeoutInMills

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

Add doc for #15011.

### Modifications

Add  `transactionBufferClientOperationTimeoutInMills` at `reference-configuration.md`

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [x] `doc-added`

### CODE REVIEW

**Comment by mattisonchao** [comment]
@momo-jun  PTAL

File: site2/docs/reference-configuration.md [hunk 1]
```diff
@@ -334,6 +334,7 @@ brokerServiceCompactionThresholdInBytes|If the estimated backlog size is greater
 |replicationConnectionsPerBroker| Max number of connections to open for each broker in a remote cluster More connections host-to-host lead to better throughput over high-latency links.  |16|
 |replicationProducerQueueSize|  Replicator producer queue size  |1000|
 |replicatorPrefix|  Replicator prefix used for replicator producer name and cursor name pulsar.repl||
+|transactionBufferClientOperationTimeoutInMills|The transaction buffer client's operation timeout in milliseconds.|3000|
```
Comment (momo-jun) [inline site2/docs/reference-configuration.md:337]: Does the new one apply to standalone Pulsar?

**Comment by momo-jun** [comment]
Another suggestion is to add "[improve]" ahead of "[doc]" in the PR title.

**Comment by mattisonchao** [comment]
> Another suggestion is to add "[improve]" ahead of "[doc]" in the PR title.

Fixed.

File: site2/docs/reference-configuration.md (hunk 1 above)
Comment (mattisonchao) [reply site2/docs/reference-configuration.md:337]: fixed.

**Review by momo-jun — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Review by RobertIndie — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**